### PR TITLE
feat: :sparkles: search for deposit id by chain

### PR DESCRIPTION
**Pull request by james-a-morris**
We can now search an arbitrary deposit id on a given chain to find all of its information. This has been very helpful to resolve deposits when looking at fills.

### CODE REVIEW

File: scripts/spokepool.ts [hunk 1]
```diff
@@ -255,6 +255,75 @@ async function fetchTxn(args: Record<string, number | string>, _signer: Signer):
   return true;
 }
 
+async function fetchDepositId(args: Record<string, number | string>): Promise<boolean> {
+  const chainId = Number(args.chainId);
+  const depositId = Number(args.depositId);
+
+  if (!utils.validateChainIds([chainId]) || Number.isNaN(depositId) || depositId < 0) {
```
Comment (nicholaspai) [inline scripts/spokepool.ts:262]: Should we replace isNaN with isInteger? or something like that

File: scripts/spokepool.ts [hunk 2]
```diff
@@ -255,6 +266,70 @@ async function fetchTxn(args: Record<string, number | string>, _signer: Signer):
...
+      [originChainId]: latestBlockOnChain.number,
+    },
+    [originChainId]
+  );
+
+  const spokeClient = spokePoolClients[originChainId];
+
+  await spokeClient.update();
+
+  const deposit = await spokeClient.findDeposit(depositId, destinationChain, depositor);
```
Comment (nicholaspai) [inline scripts/spokepool.ts:309]: nit: if undefined, should we throw a detailed error recommending caller to set longer `hours`?

**Review by nicholaspai — APPROVED** [review]
https://github.com/across-protocol/relayer-v2/pull/1119/files#r1428917153

File: scripts/spokepool.ts (hunk 2 above)
Comment (james-a-morris) [reply scripts/spokepool.ts]: Good call - done

File: scripts/spokepool.ts (hunk 1 above)
Comment (james-a-morris) [reply scripts/spokepool.ts]: Good call - done

**Comment by pxrl** [comment]
This is a really useful feature that we've needed for a long time. I'm not a big fan of introducing the kitchen sink HubPoolClient and SpokePoolClient dependencies into this script because it's intended to be small and lightweight, and adding them starts to bog it down. I think there's a cleaner way of achieving this without adding them by just relying on the filtering functions of the SpokePool contract instance. I also think it'd be better placed as part of the existing `fetch` command - i.e. `fetch --chainId [--txnHash | --depositId]`.

**Comment by pxrl** [comment]
> This is a really useful feature that we've needed for a long time. I'm not a big fan of introducing the kitchen sink HubPoolClient and SpokePoolClient dependencies into this script because it's intended to be small and lightweight, and adding them starts to bog it down. I think there's a cleaner way of achieving this without adding them by just relying on the filtering functions of the SpokePool contract instance. I also think it'd be better placed as part of the existing `fetch` command - i.e. `fetch --chainId [--txnHash | --depositId]`.

fwiw I took this PR and hacked on it a bit to show an example: https://github.com/across-protocol/relayer-v2/pull/1122

I think the above is cleaner in a few ways, primarily because it doesn't lump this script with having to know about the HubPool and SpokePoolClient types.

One counter-argument might be that it can limit the set of providers that can be used because the block range for the filter query is so large. It works fine with Infura and Arbitrum though, and I'm probably OK with that given that it keeps the code lighter.

**Review by mrice32 — COMMENTED** [review]
@james-a-morris @pxrl can we close one of the PRs in favor of the other?

For reference, this is the other one: https://github.com/across-protocol/relayer-v2/pull/1122

**Pull request closed without merging**